mailpoet/lib/Cron/Workers/SendingQueue/SendingQueue.php

```diff
@@ -248,20 +247,32 @@ private function processSending(ScheduledTaskEntity $task, int $timer): void {
           $this->scheduledTasksRepository->flush();
           return;
         }
-        $foundSubscribers = empty($foundSubscribersIds) ? [] : SubscriberModel::whereIn('id', $foundSubscribersIds)
-          ->whereNull('deleted_at')
-          ->findMany();
+        $foundSubscribers = empty($foundSubscribersIds) ? [] : $this->subscribersRepository->findBy(['id' => $foundSubscribersIds, 'deletedAt' => null]);
       } else {
         // No segments = Welcome emails or some Automatic emails.
         // Welcome emails or some Automatic emails use segments only for scheduling and store them as a newsletter option
-        $foundSubscribers = SubscriberModel::whereIn('id', $subscribersToProcessIds);
-        $foundSubscribers = $newsletter->type === NewsletterEntity::TYPE_AUTOMATION_TRANSACTIONAL ?
-          $foundSubscribers->whereNotEqual('status', SubscriberModel::STATUS_BOUNCED) :
-          $foundSubscribers->where('status', SubscriberModel::STATUS_SUBSCRIBED);
-        $foundSubscribers = $foundSubscribers
-          ->whereNull('deleted_at')
-          ->findMany();
-        $foundSubscribersIds = SubscriberModel::extractSubscribersIds($foundSubscribers);
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+
+        $queryBuilder->select('s')
+          ->from(SubscriberEntity::class, 's')
+          ->where('s.id IN (:subscriberIds)')
+          ->setParameter('subscriberIds', $subscribersToProcessIds)
+          ->andWhere('s.deletedAt IS NULL');
+
+        if ($newsletterEntity->getType() === NewsletterEntity::TYPE_AUTOMATION_TRANSACTIONAL) {
+          $queryBuilder->andWhere('s.status != :bouncedStatus')
+            ->setParameter('bouncedStatus', SubscriberEntity::STATUS_BOUNCED);
+        } else {
+          $queryBuilder->andWhere('s.status = :subscribedStatus')
+            ->setParameter('subscribedStatus', SubscriberEntity::STATUS_SUBSCRIBED);
+        }
+
+        $foundSubscribers = $queryBuilder->getQuery()->getResult();
+        $foundSubscribersIds = array_filter(
+          array_map(function(SubscriberEntity $subscriber) {
+            return (!empty($subscriber->getId())) ? $subscriber->getId() : false;
+          }, $foundSubscribers)
+        );
```

Comment:
I'm not sure why the `array_filter` would be needed here. Seems like all subscriber entities fetched by a query need to have an ID. This code would also exclude ID `0`, but I don't think we have that. I'll investigate this further.

Additionally, it's safer to wrap `array_filter` with `array_valus` to avoid gaps in indexes (and thus associative array instead of list).

Reply:
Ahh, found that this was extracted from here: https://github.com/mailpoet/mailpoet/blob/b43eac8af33b341f104a9de376984798112a23f3/mailpoet/lib/Models/Subscriber.php#L643-L649

I think it's safe to remove the check.